experimental/dds/tree2/src/util/utils.ts

```diff
@@ -247,6 +247,16 @@ export function isJsonObject(
 	return typeof value === "object" && value !== null && !Array.isArray(value);
 }
 
+export function assertValidRangeIndices(
+	startIndex: number,
+	endIndex: number,
+	array: { readonly length: number },
+) {
+	assert(endIndex > startIndex, "Range indices are malformed.");
```

Comment:
Should endIndex === startIndex be valid as a 0-length range (at least for some scenarios)?